examples/two_dimensional_mountain_wave.jl

```diff
@@ -0,0 +1,386 @@
+# representation:
+
+# Analytical profile on a high-resolution grid:
+analytical_grid = RectilinearGrid(CPU(), size=500, x=(-30e3, 30e3), topology=(Periodic, Flat, Flat))
+h_analytical = Field{Center, Nothing, Nothing}(analytical_grid)
+set!(h_analytical, hill)
+
+# Discretized profile as represented in the model:
+h_model = grid.immersed_boundary.bottom_height
+
+fig_terrain = Figure(size=(900, 400))
+ax_terrain = Axis(fig_terrain[1, 1],
+                  xlabel = "x (m)",
+                  ylabel = "Height (m)",
+                  title = "Schär Mountain Profile")
+lines!(ax_terrain, h_analytical, linewidth = 1, color = :black, 
+       label = "Analytical")
+lines!(ax_terrain, h_model, linewidth = 2, color = :brown, linestyle = :dash,
+       label = "Model")
+
+# band! requires arrays, not Fields
+x_ana = xnodes(h_analytical)
+h_ana = interior(h_analytical, :, 1, 1)
+band!(ax_terrain, x_ana, zeros(length(x_ana)), h_ana, color = (:brown, 0.2))
+xlims!(ax_terrain, -30e3, 30e3)
+axislegend(ax_terrain, position = :rt)
+
+save("mountain_wave_terrain.png", fig_terrain)
+fig_terrain
+
+# ## Rayleigh damping layer
+#
+# A sponge layer at the top of the domain prevents spurious wave reflections from
+# the upper boundary. The damping follows a Gaussian profile centered at the domain top:
+#
+# ```math
+# \mathcal{S}(\rho w) = -\omega \, \exp\left(-\frac{(z - z_0)^2}{2 \Delta z^2}\right) \rho w
+# ```
+#
+# where ``\omega = 1/60 \, {\rm s}^{-1}`` is the relaxation rate and ``\Delta z = L_z/2``.
+
+@inline gaussian_mask(z, p) = exp(-(z - p.z0)^2 / 2p.dz^2)
+
+@inline function sponge(i, j, k, grid, clock, params, ρc, lz)
+    z = znode(k, grid, lz)
+    ω = params.ω
+    m = gaussian_mask(z, params)
+    return @inbounds -ω * m * ρc[i, j, k]
+end
+
+@inline ρw_sponge(i, j, k, grid, clock, model_fields, params) =
+    sponge(i, j, k, grid, clock, params, model_fields.ρw, Face())
+
+sponge_params = (z0 = grid.Lz, dz = grid.Lz / 2, ω = 1/60)
+ρw_forcing = Forcing(ρw_sponge, discrete_form=true, parameters=sponge_params)
+
+# ## Model initialization
+#
+# Create the atmosphere model with the anelastic formulation, 5th-order WENO advection,
+# and the Rayleigh damping layer:
+reference_state = ReferenceState(grid, constants, surface_pressure=p₀, potential_temperature=θ₀)
+formulation = AnelasticFormulation(reference_state)
+
+advection = WENO(order=5)
+model = AtmosphereModel(grid; formulation, advection, forcing=(; ρw=ρw_forcing))
+
+# ## Initial conditions
+#
+# Initialize with the constant-``N`` stratification and uniform background wind.
+# The potential temperature profile that maintains constant Brunt–Väisälä frequency is:
+#
+# ```math
+# \theta(z) = \theta_0 \exp\left(\frac{N^2 z}{g}\right)
+# ```
+
+θᵢ(x, z) = θ₀ * exp(N² * z / g)
+set!(model, θ = θᵢ, u = U)
+
+# ## Simulation
+#
+# Run for 2 hours with a fixed time step. A small time step is required for numerical stability with this test case:
```

Comment:
simulation blows up for me; even when I changed to 

```Julia
simulation = Simulation(model; Δt=0.5, stop_time=2hours)
conjure_time_step_wizard!(simulation, cfl=0.7)
```

I don't know why... :(